utils/build-script-impl

```diff
@@ -926,138 +1067,163 @@ if [[ ! "${SKIP_BUILD_SWIFTPM}" ]] ; then
-            if [[ ! "${SKIP_TEST_TVOS_HOST}" ]] ; then
-                test_host_only=1
-            else
-                test_this_target=
-            fi
-            build_benchmark_this_target=$(not ${SKIP_BUILD_TVOS_DEVICE})
-            ;;
-        appletvsimulator-*)
-            build_for_this_target=$(not ${SKIP_BUILD_TVOS_SIMULATOR})
-            test_this_target=$(not ${SKIP_TEST_TVOS_SIMULATOR})
-            ;;
-        watchos-*)
-            build_for_this_target=$(not ${SKIP_BUILD_WATCHOS_DEVICE})
-            if [[ ! "${SKIP_TEST_WATCHOS_HOST}" ]] ; then
-                test_host_only=1
-            else
+function calculate_targets_for_host() {
+    local host=$1
+    
+    SWIFT_STDLIB_TARGETS=()
+    SWIFT_SDKS=()
+    SWIFT_BENCHMARK_TARGETS=()
+    SWIFT_RUN_BENCHMARK_TARGETS=()
+    SWIFT_TEST_TARGETS=()
+
+    # Get the list of Target platforms for the Host
+    local stdlib_targets=(`echo $(get_stdlib_targets_for_host ${host})`)
+
+    for stdlib_deployment_target in "${stdlib_targets[@]}"; do
+        local swift_sdk=
+        local is_in_build_list=$(should_build_stdlib_target ${stdlib_deployment_target} ${host})
+        local build_for_this_target=1
+        local test_this_target=1
+        local test_host_only=
+        local build_benchmark_this_target=
+        local test_benchmark_this_target=
+
+        case ${stdlib_deployment_target} in
+            linux-*)
+                swift_sdk="LINUX"
+                build_for_this_target=$(not ${SKIP_BUILD_LINUX})
+                test_this_target=$(not ${SKIP_TEST_LINUX})
+                ;;
+            freebsd-*)
+                swift_sdk="FREEBSD"
+                build_for_this_target=$(not ${SKIP_BUILD_FREEBSD})
+                test_this_target=$(not ${SKIP_TEST_FREEBSD})
+                ;;
+            cygwin-*)
+                swift_sdk="CYGWIN"
+                build_for_this_target=$(not ${SKIP_BUILD_CYGWIN})
+                test_this_target=$(not ${SKIP_TEST_CYGWIN})
+                ;;
+            macosx-*)
+                swift_sdk="OSX"
+                build_for_this_target=$(not ${SKIP_BUILD_OSX})
+                test_this_target=$(not ${SKIP_TEST_OSX})
+                build_benchmark_this_target=$(not ${SKIP_BUILD_OSX})
+                test_benchmark_this_target=$(not ${SKIP_BUILD_OSX})
+                ;;
+            iphoneos-*)
+                swift_sdk="IOS"
+                build_for_this_target=$(not ${SKIP_BUILD_IOS_DEVICE})
+                if [[ ! "${SKIP_TEST_IOS_HOST}" ]] ; then
+                    test_host_only=1
+                else
+                    test_this_target=
+                fi
+                build_benchmark_this_target=$(not ${SKIP_BUILD_IOS_DEVICE})
+                
+                # Never build iOS armv7s benchmarks.
+                if [[ "${stdlib_deployment_target}" == "iphoneos-armv7s" ]]; then
+                    build_benchmark_this_target=
+                fi
```

Comment:
Hmm, so we weren't explicitly building for `armv7s` before, but our CMake configuration makes all iOS architectures depend on each other so it just ends up getting built anyway. I wouldn't really like to add this back; it basically relies on a bug/accidental behaviour.

Unfortunately, the cleanest way to skip benchmarks for one particular stdlib target but build others in the same SDK right now is to hardcode it in the build script.